# T20 — First exploration: Hillstrom response rates

Exploring response (visit) rates by treatment arm before committing
the final figure. The committed chart is generated by
`scripts/figures/response_by_arm.py`, not exported from here.

In [1]:
import pandas as pd

hill = pd.read_csv("../data/raw/hillstrom.csv")
hill.head()

,recency,history_segment,history,mens,womens,zip_code,newbie,channel,segment,visit,conversion,spend
0,10,2) $100 - $200,142.44,1,0,Surburban,0,Phone,Womens E-Mail,0,0,0.0
1,6,3) $200 - $350,329.08,1,1,Rural,1,Web,No E-Mail,0,0,0.0
2,7,2) $100 - $200,180.65,0,1,Surburban,1,Web,Womens E-Mail,0,0,0.0
3,9,5) $500 - $750,675.83,1,0,Rural,1,Web,Mens E-Mail,0,0,0.0
4,2,1) $0 - $100,45.34,1,0,Urban,0,Web,Womens E-Mail,0,0,0.0


In [2]:
hill["segment"].value_counts()

segment
Womens E-Mail    21387
Mens E-Mail      21307
No E-Mail        21306
Name: count, dtype: int64

In [3]:
import subprocess
import sys
from pathlib import Path

cwd = Path.cwd()
project_root = cwd.parent if cwd.name == "notebooks" else cwd
script_path = project_root / "scripts" / "figures" / "response_by_arm.py"

result = subprocess.run(
    [sys.executable, str(script_path)],
    capture_output=True,
    text=True,
    cwd=project_root,
    check=False,
)

print(result.stdout)
if result.returncode != 0:
    print("FAILED with exit code", result.returncode)
    print(result.stderr)

Mens E-Mail     n= 21307  visit rate=18.2757%  95% CI=[17.7734%, 18.8296%]
Womens E-Mail   n= 21387  visit rate=15.1400%  95% CI=[14.6678%, 15.6124%]
No E-Mail       n= 21306  visit rate=10.6167%  95% CI=[10.2178%, 11.0485%]

Saved: reports\figures\response_by_arm.png

What a response model would conclude:
The Mens E-Mail arm has the highest visit rate (18.28%), so a response model would rank those customers highest and target them.

Why that does not tell us who to send the email to:
This chart shows who responded, not whose response was CAUSED by the email. The No E-Mail (control) arm still shows a 10.62% visit rate with no email at all -- so part of every arm's rate would have happened regardless of treatment. Only the gap against this control arm, not the raw rate, estimates the email's actual effect.



The chart below is generated by running:



![response by arm](../reports/figures/response_by_arm.png)

In [4]:
from pathlib import Path

print("Current directory:", Path.cwd())
print("Contents:", list(Path.cwd().iterdir()))

Current directory: c:\Projects\Depi\DEPI_Final Project\netlift\notebooks
Contents: [WindowsPath('c:/Projects/Depi/DEPI_Final Project/netlift/notebooks/.gitkeep'), WindowsPath('c:/Projects/Depi/DEPI_Final Project/netlift/notebooks/01_hillstrom_exploration.ipynb')]


**What a response model would conclude:** the Mens E-Mail arm has the highest visit rate (18.28%, against 10.62% with no email), so a response model would rank customers like those in that arm highest and target them first.

**Why that does not tell us who to send the email to:** each bar is an average over everyone in the arm, so it says nothing about which individual customers changed behaviour because of the email, and some of those who visited would have visited anyway (10.62% did with no email at all); the gap between arms estimates the average effect, but sending to the customers most likely to visit is not the same as sending to the customers the email actually persuades.